### PR TITLE
[fix 5496][master] fix npe

### LINKED ISSUE #5496: [Bug][Master] NPE occurs in complement operation
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.server.master.runner.MasterExecThread.executeComplementProcess(MasterExecThread.java:336) ~[classes/:na]
	at org.apache.dolphinscheduler.server.master.runner.MasterExecThread.run(MasterExecThread.java:219) ~[classes/:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_291]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_291]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_291]

复线步骤：
1、运行时点击补数
2、调度日期选择多天，至少俩天
3、然后点运行
4、master节点报npe，如上

排查发现：
问题出现在MasterExecThread： 470行
这里查询完后导致processDefinition属性变为null。后面使用时报npe了
image

branch: [dev]

### PR DESCRIPTION
**Pull request by caoping6**
## Brief change log
fix npe

this fix #5496

### CODE REVIEW

**Comment by CalvinKirs** [comment]
Can you describe in detail what problem this PR solves?
BTW, please use English.

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java [hunk 1]
```diff
@@ -323,6 +324,8 @@ private void executeComplementProcess() throws Exception {
             }
             // flow end
             // execute next process instance complement data
+            ProcessDefinition processDefinition = processService.findProcessDefineById(processDefinitionId);
```
Comment (brave-lee) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:327]: Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java [hunk 2]
```diff
@@ -930,7 +933,7 @@ private void runProcess() {
             if (!sendTimeWarning && checkProcessTimeOut(processInstance)) {
                 processAlertManager.sendProcessTimeoutAlert(processInstance,
                         processService.findProcessDefinition(processInstance.getProcessDefinitionCode(),
-                        processInstance.getProcessDefinitionVersion()));
+                                processInstance.getProcessDefinitionVersion()));
```
Comment (brave-lee) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java]: Please use unified codestyle

**Comment by CalvinKirs** [comment]
Hi, how is it getting?

**Comment by davidzollo** [comment]
hi caoping6,
you have already done a good job when you submitted your first PR, it's the fist step for you guys to join the Contributor Community. 
I think you can contact me (wechat: taskflow or mail: lidongdai@apache.org) when you need any help

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java (hunk 1 above)
Comment (davidzollo) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:327]: > Thanks for your contribution. Why is it need set processDefinition here? Can you describe the problems you encounter in detail

I also have the same question, ^-^

Comment (echohlne) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java:327]: @dailidong @JinyLeeChina  @caoping6 
It seems a same problem described in #5909
But it‘s better not to  modify at here.
I personally think that the problem comes in  [MasterExecThread.java#L977](https://github.com/apache/dolphinscheduler/blob/1887bde1ebf8249de153890e78d449582e3eaf4a/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/MasterExecThread.java#L977)
aftert this:
`processInstance = processService.findProcessInstanceById(processInstance.getId());`
the `processDefinition` variable in `ProcessInstance` woule be null.
```java
// ProcessInstance.java
    /**
     * process definition structure
     */
    @TableField(exist = false)
    private ProcessDefinition processDefinition;
```

**Comment by caishunfeng** [comment]
The latest code had been changed, so this pr will be closed.

**Pull request closed without merging**